cmd/signer/signer_test.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/crypto"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/server/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSigner(t *testing.T) {
+	t.Run("incorrect orgs", func(t *testing.T) {
+		for _, arg := range []string{
+			"",
+			"-data=\"{\\\"userID\\\":\\\"admin\\\"}\"",
+			"-privatekey=../../deployment/sample/crypto/admin/admin.key",
+		} {
+			out, err := exec.Command("go", "run", "signer.go", arg).CombinedOutput()
+			require.NoError(t, err)
+
+			expectedOut := help + "\n  -data string\n    \tjson data to be signed. All double quotes must be escaped." +
+				" An example json data is \"{\\\"userID\\\":\\\"admin\\\"}\"\n  -privatekey string\n    \tpath to the private key" +
+				" to be used for adding a digital signature\n"
+			require.Equal(t, expectedOut, string(out))
+		}
+	})
+
+	t.Run("correct org", func(t *testing.T) {
+		tempDir, err := ioutil.TempDir("/tmp", "crypto")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			if err := os.RemoveAll(tempDir); err != nil {
+				t.Errorf("error while removing test directory: %v", err)
+			}
+		})
+
+		adminCert, adminKey := createAdminCreds(t, tempDir)
+		args := []string{
+			"run",
+			"signer.go",
+			"-data=\"{\\\"user_id\\\":\\\"admin\\\"}\"",
+			"-privatekey=" + adminKey,
+		}
+
+		out, err := exec.Command("go", args...).Output()
+		require.NoError(t, err)
+
+		b, err := ioutil.ReadFile(adminCert)
+		require.NoError(t, err)
+		bl, _ := pem.Decode(b)
+		require.NotNil(t, bl)
+		verifier, err := crypto.NewVerifier(bl.Bytes)
+		require.NoError(t, err)
+
+		signature, err := base64.StdEncoding.DecodeString(string(out))
+		require.NoError(t, err)
+		require.NoError(t, verifier.Verify([]byte("{\"user_id\":\"admin\"}"), signature))
```

Comment:
```
adminCert, adminKey := createAdminCreds(t, tempDir)
		data := "\"{\"user_id\":\"admin\"}\""
		args := []string{
			"run",
			"signer.go",
			"-data=" + data,
			"-privatekey=" + adminKey,
		}

		out, err := exec.Command("go", args...).Output()
		require.NoError(t, err)

		b, err := ioutil.ReadFile(adminCert)
		require.NoError(t, err)
		bl, _ := pem.Decode(b)
		require.NotNil(t, bl)
		verifier, err := crypto.NewVerifier(bl.Bytes)
		require.NoError(t, err)

		signature, err := base64.StdEncoding.DecodeString(string(out))
		require.NoError(t, err)
		require.NoError(t, verifier.Verify([]byte(data), signature))
```
double quotes are not the same in sign and verify...

Reply:
Thanks for finding this. I got confused with how the json marshalled bytes would look like.